docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1494 @@
+  no-overlay mode but do not want to expose pod IPs externally, this behavior is
+  not ideal. To address this, users will be able to enable SNAT for
+  pod-to-external traffic. This will be configured via a new `outboundSNAT`
+  field in the `ClusterUserDefinedNetwork` CRD for CUDNs, and the
+  `outbound-snat` flag in the `[no-overlay]` section of the configuration file
+  for the default network.
+
+* For east-west traffic:
+  * Intra-node traffic remains unchanged, handled by the local OVN bridge.
+  * Inter-node traffic will follow the same path as north-south traffic, routed
+    via the underlay. Pod-to-pod, Pod-to-clusterIP traffic within the same
+    network will be forwarded by the provider network without encapsulation or
+    NAT. Pod-to-node egress traffic will be SNATed to the node IP at the source
+    node to allow a pod to access nodePort services in a different UDN.
+
+### Configuration File and Flags
+
+For the default network transport, the new flag `transport` will be added to
+the `[default]` section of the configuration file:
+
+* The `transport` flag accepts either `no-overlay` or
+  `geneve`, defaulting to `geneve`. Setting it to `no-overlay` configures the
+  default network to operate in no-overlay mode. More overlay transport
+  methods can be supported in the future. If
+  `transport=no-overlay` is set, but
+  no `RouteAdvertisements` CR is configured for advertising the default network,
+  ovnkube-cluster-manager will emit an event and log an error message.
+
+For the default network in no-overlay mode, the new flags `outbound-snat` and
+`routing` will be added to the `[no-overlay]` section of the configuration file:
+
+* The `outbound-snat` flag is a string flag to configure the SNAT
+  behavior for outbound traffic from pods on the default network. Supported
+  values are `enable` or `disable`. It is required when `transport=no-overlay`.
+
+  * `enable`: SNATs pod-to-external traffic to the node IP. This is useful when
+    pod IPs are not routable on the external network.
+  * `disable`: Does not SNAT pod-to-external traffic. This requires pod IPs to
+    be routable on the external network.
+
+  In no-overlay mode, pod-to-remote-pod traffic is never SNATed, regardless of
+  this flag's setting. However, traffic to remote nodes, the Kubernetes API
+  server, and DNS is always SNATed.
+
+* The `routing` flag is a string flag to configure
+  whether the pod network routing configuration is managed by OVN-Kubernetes or
+  users. Supported values are `managed` or `unmanaged`. It is required when
+  `transport=no-overlay`.
+
+  * `managed`: OVN-Kubernetes manages the routing for the no-overlay network.
+    OVN-Kubernetes will generate the BGP configurations (RouteAdvertisements and
+    FRRConfiguration) to advertise the pod subnets, and set up BGP sessions
+    between nodes. In the managed routing mode, pod subnets will be advertised
+    to the default VRF of each node for both the default network and CUDNs.
```

Comment:
so this "managed" config flag enables managed for all the CUDNs as well? so it is not configurable in the CUDN?

Reply:
This flag is for the default network only. CUDN gets its own flag in the CRD. This is a general statement for the managed mode. I should move this line somewhere else.